pyV2DL3/eventdisplay/fillRESPONSE.py

```diff
@@ -37,15 +38,50 @@ def print_logging_info(irf_to_store, camera_offsets, pedvar, zenith):
 
 
 def check_parameter_range(par, par_irf, par_name):
-    """Check that coordinates are in range of provided IRF"""
+    """Check that coordinates are in range of provided IRF and whether extrapolation is to be done
+       0. checks if command line parameter force_extrapolation is given. If given,
+          the extrapolation will happen when parameter is outside IRF range. If parameter is
+          within IRF range, it works as normal. Default is False.
+       1. Further checks for fuzzy boundary (parameter close to boundary value).
+          If fuzzy boundary is within a given tolerance then IRF is interpolated for
+          at bounary value. Default is 0.0 tolerance.
+    """
 
     logging.info(
-        "\t{0} range of a given IRF: {1:.1f} - {2:.1f}".format(
+        "\t{0} range of a given IRF: {1:.2f} - {2:.2f}".format(
             par_name, np.min(par_irf), np.max(par_irf)
         )
     )
+    clk = click.get_current_context()
+    force_extrapolation = clk.params["force_extrapolation"]
+    fuzzy_boundary = clk.params["fuzzy_boundary"]
     if np.all(par_irf < par) or np.all(par_irf > par):
```

Comment:
Could we change this part to have not 4 blocks intended which makes it hard to read? Maybe easier to put some of the code in functions with very descriptive names?

Reply:
Thank you for the reviewing this PR. I am not sure whether I understand this comment. Can you please elaborate a bit what changes are you asking to implement?

Reply:
It is hard to read as there are too many levels of indentation.